app/assets/javascripts/character_count.js

```diff
@@ -4,57 +4,60 @@ $(() => {
     el.removeClass(icons.join(' ')).addClass(icon);
   };
 
-  $(document).on('keyup change paste', '[data-character-count]', ev => {
+  const setCounterState = (el, state) => {
+    if(state == 'info') {
+      el.removeClass('has-color-yellow-700 has-color-red-500').addClass('has-color-primary');
+    } else if(state === 'warning') {
+      el.removeClass('has-color-red-500 has-color-primary').addClass('has-color-yellow-700');
+    } else if(state === 'error') {
+      el.removeClass('has-color-yellow-700 has-color-primary').addClass('has-color-red-500');
+    } else {
+      el.removeClass('has-color-red-500 has-color-yellow-700 has-color-primary');
+    }
```

Comment:
style: spaces before parens, new blocks on new lines:  
lint: strict equality checks (`===`) in JS unless you're dealing with `null/undefined`:

```js
if (a === 'b') {
  doAThing();
}
else if (a === 'c') {
  doAnotherThing();
}
```

There's a number of these across the PR - worth a double-check for them.

Reply:
@ArtOfCode- still a draft, haven't bothered with linting it properly (my personal personal preference is same-line, so that's probably why), will do when I have enough time to spare to finalize it. As for the strict equality - yeah, seems like a typo, I don't think I've ever intentionally used lax equality.